src/js/components/DataTable/doc.js

```diff
@@ -83,6 +83,17 @@ export const doc = DataTable => {
       `Cell border. You can set the border per context by passing an
       object with keys for 'heading', 'body', and/or 'footer'.`,
     ),
+    columnGroups: PropTypes.arrayOf(
```

Comment:
I'm thinking `groupHeader` or `headerGroups` will be a more straight forward naming.